### DISCUSSION TITLE
transfer learning

### DISCUSSION THREAD

**Discussion by nurislam690206** [ultralytics/ultralytics#18519] [Q&A]
Is it possible to use feature extraction and fine-tuning with YOLO11?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @nurislam690206, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug it.

If this is a custom training ❓ Question, like in your case regarding feature extraction and fine-tuning with YOLO11, please provide as much information as possible, including dataset image examples, relevant code snippets, or training configurations, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response, but no worries! An Ultralytics engineer will assist you here soon to provide further clarification 😊

**Comment by glenn-jocher** [comment]
@nurislam690206 yes, YOLO11 supports transfer learning, including feature extraction and fine-tuning, which can be achieved by training the model on your custom dataset using pre-trained weights. For guidance, refer to the [Train YOLO11 documentation](https://docs.ultralytics.com/modes/train/).

**Reply by nurislam690206** [reply]
Thank you for your reply.

**Reply by nurislam690206** [reply]
Please check the images
![Capture2](https://github.com/user-attachments/assets/89f58959-6dec-431b-9a62-c8d8cc5f3184)
![Capture3](https://github.com/user-attachments/assets/44113d91-7b26-479b-8c53-ab3465aa8038)
![Capture](https://github.com/user-attachments/assets/b0916216-c729-412c-b8fe-d1115ab80476)
![Capture1](https://github.com/user-attachments/assets/41777703-ea18-4979-92be-acd94fc4286a)
what's the wrong?

**Reply by nurislam690206** [reply]
Dataset not found , missing path [D:\Dataset\datasets\YOLO11\data_imageNet.yaml](file:///D:/Dataset/datasets/YOLO11/data_imageNet.yaml)
![image](https://github.com/user-attachments/assets/7c2e4523-43bc-45e3-81c9-c43baa04a4f0)

in this directory, I have 'data_imageNet.yaml' file

**Reply by glenn-jocher** [reply]
The error suggests that the path to your `data_imageNet.yaml` file is not being correctly resolved. Ensure the path is accurate and accessible. If the issue persists, verify that the YAML file is formatted correctly as per the [Ultralytics YOLO dataset format guidelines](https://docs.ultralytics.com/datasets/detect/).

**Reply by nurislam690206** [reply]
[Custom_yaml file.txt](https://github.com/user-attachments/files/18312607/Custom_yaml.file.txt)
[New DOCX 文档.docx](https://github.com/user-attachments/files/18312713/New.DOCX.docx)

 please check?

**Reply by glenn-jocher** [reply]
@nurislam690206 it looks like the issue may be with the structure or content of your `data_imageNet.yaml` file. Please ensure the file follows the correct [Ultralytics YOLO dataset YAML format](https://docs.ultralytics.com/datasets/detect/) and that the paths specified within it are accurate and accessible. If the problem persists, please share any specific errors or issues encountered for further assistance.

**Reply by nurislam690206** [reply]
I solved the problem...Thank you so much for your help👌

**Reply by glenn-jocher** [reply]
You're welcome! Glad to hear the issue is resolved. The credit goes to the YOLO community and the Ultralytics team for their resources. Let us know if you need further assistance!
